libp2p-swarm/src/transport.rs

```diff
@@ -579,14 +579,12 @@ impl<'a, T, C> UpgradedNode<T, C>
 					(n, <Bytes as PartialEq>::eq, t)
 				}
 				let iter = upgrade.protocol_names().map(iter_map);
-				let negotiated = multistream_select::listener_select_proto(connection, iter)
-					.map_err(|err| panic!("{:?}", err));      // TODO:
+				let negotiated = multistream_select::listener_select_proto(connection, iter);
 				negotiated.map(move |(upgrade_id, conn)| (upgrade_id, conn, upgrade, client_addr))
-					.map_err(|_| panic!())    // TODO:
+					.map_err(|err| IoError::new(IoErrorKind::Other, err))
```

Comment:
Style nit: I prefer `use std::io;` to `use std::io::Error as IoError;`, but that could be argued as personal taste.